modules/heat_transfer/src/kernels/HeatConductionTimeDerivative.C

```diff
@@ -33,13 +37,21 @@ HeatConductionTimeDerivative::validParams()
    */
   params.addParam<MaterialPropertyName>(
       "density_name", "density", "Property name of the density material property");
+  params.addParam<MaterialPropertyName>(
+      "density_name_dT",
+      "Name of material property for the derivative of the density with respect to the variable.");
   return params;
 }
 
 HeatConductionTimeDerivative::HeatConductionTimeDerivative(const InputParameters & parameters)
   : TimeDerivative(parameters),
     _specific_heat(getMaterialProperty<Real>("specific_heat")),
-    _density(getMaterialProperty<Real>("density_name"))
+    _specific_heat_dT(isParamValid("specific_heat_dT")
+                          ? &getMaterialProperty<Real>("specific_heat_dT")
+                          : nullptr),
+    _density(getMaterialProperty<Real>("density_name")),
+    _density_dT(isParamValid("density_name_dT") ? &getMaterialProperty<Real>("density_name_dT")
```

Comment:
Any reason not to use the derivative material property name interface here? I'd really like to avoid using manual derivative property names.

Reply:
I'm kinda with you, but this is a consequence of the extensive use of `HeatConductionMaterial` with functions to describe thermal conductivity and heat conduction. Not a terrible way to do things. This retroactively allows use of the exact same work flow with the dT. If you want to get rid of this, then we should deprecate `HeatConductionMaterial`. Not sure I see why this shouldn't use the same technique used for thermal conductivity.